UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -223,6 +232,13 @@ private async Task<string> SelectLatestPublishedDateBatch(List<Products> product
             return uri;
         }
```

Comment:
I realise that you have only added to this method but the cyclic complexity has grown with the latest additions and this is probably ready to be refactored to make it more readable/understandable.

Is it possible to refactor this to reduce the complexity?

Reply:
Yes refactored the method.